chef/cookbooks/monasca/recipes/log_agent.rb

```diff
@@ -0,0 +1,113 @@
+# Copyright 2017 FUJITSU LIMITED
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package "monasca-log-agent"
+
+log_agent_settings = node[:monasca][:log_agent]
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+log_agent_dimensions = {
+  service: "monitoring",
+  hostname: node["hostname"]
+}
+# TODO(trebskit) actually it should be retrieved from keystone
+log_agent_settings["monasca_log_api_url"] = "http://www.example.org"
+
+### create user for monasca-log-agent
+register_auth_hash = {
+  user: keystone_settings["admin_user"],
+  password: keystone_settings["admin_password"],
+  tenant: keystone_settings["admin_tenant"]
+}
+
+keystone_register "monasca-log-agent wakeup keystone" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  action :wakeup
+end
+
+keystone_register "register monasca-log-agent user" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  user_name keystone_settings["service_user"]
+  user_password keystone_settings["service_password"]
+  tenant_name keystone_settings["service_tenant"]
+  action :add_user
+end
+
+keystone_register "give monasca-log-agent user access" do
+  protocol keystone_settings["protocol"]
+  insecure keystone_settings["insecure"]
+  host keystone_settings["internal_url_host"]
+  port keystone_settings["admin_port"]
+  auth register_auth_hash
+  user_name keystone_settings["service_user"]
+  tenant_name keystone_settings["service_tenant"]
+  role_name "admin"
```

Comment:
Again, if the above holds true:

* The tenant_name should point to a dedicated tenant for the log user (e.g. `cmm` if we were to use the same tenant name as the Ansible installer) which needs to be created first (since this tenant is used by both the agent and the server it's probably best to introduce a `common.rb` recipe that creates it and is run at the start at both the start of both the agent and server Crowbar roles).
* The role should be `"cmm-agent"` if this is to be an unprivileged user sending logs and monitoring events (maybe a good idea to introduce an attribute for the role, too).